### PR TITLE
Add per-miner RSR

### PR DESCRIPTION
**Pull request by pyropy**
- Adds per-miner RSR endpoint
- Closes issue #106

### CODE REVIEW

File: stats/test/handler.test.js [hunk 1]
```diff
@@ -674,6 +674,83 @@ describe('HTTP request handler', () => {
...
+      )
+      await assertResponseStatus(res, 200)
+      const stats = (/** @type {{ day: string, success_rate: number }[]} */
+        await res.json()
+      )
+      assert.deepStrictEqual(stats, [
+        { day: '2024-01-10', success_rate: 5 / 10, total: '10', successful: '5' },
+        { day: '2024-01-20', success_rate: 1 / 10, total: '10', successful: '1' }
+      ])
+    })
```
Comment (bajtos) [inline stats/test/handler.test.js:752]: It's great that you considered various scenarios to cover with tests 👍🏻 

The test code looks good but is also a bit repetitive. Let's find a way how to cover multiple scenarios in fewer test cases.

I think a dataset like the following covers all three cases, WDYT?

```js
      // before the range
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-09', minerId: 'f1one', total: 10, successful: 1 })
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-09', minerId: 'f1two', total: 100, successful: 20 })
      // in the range
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-20', minerId: 'f1one', total: 10, successful: 1 })
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-20', minerId: 'f1two', total: 200, successful: 60 })
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-10', minerId: 'f1one', total: 10, successful: 5 })
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-10', minerId: 'f1two', total: 100, successful: 50 })
      // after the range
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-21', minerId: 'f1one', total: 30, successful: 1 })
      await givenRetrievalStats(pgPools.evaluate, { day: '2024-01-21', minerId: 'f1two', total: 300, successful: 60 })
```

The trick is to realise that `assert.deepStrictEqual(stats,` checks multiple conditions:
- `stats` contain only items for the selected miner
- `stats` contain only items from the selected date range
- `stats` contain items ordered by `day` (ascending)

Why is this important: when the tests are chatty and verbose, it's more work to update them. For example, I envision adding another field counting the number of successful retrievals over HTTP (not Graphsync; see https://github.com/space-meridian/roadmap/issues/192). In your proposal, there are three tests to update. In my proposal, there is only one test to update.

**Review by bajtos — CHANGES_REQUESTED** [review]
Wow, this was quick!

The changes look great at the high level; let's improve a few details now.

File: stats/test/handler.test.js (hunk 1 above)
Comment (pyropy) [reply stats/test/handler.test.js:752]: Makes sense!

Comment (pyropy) [reply stats/test/handler.test.js]: Test have been updated, thanks for pointing this out.

**Review by bajtos — APPROVED** [review]
LGTM 👍🏻 

What's next:
- Merge the main branch into your feature branch (e.g. using the button in the GitHub UI)
- Wait for all CI checks to finish
- Merge the pull request 🚀

**Review by juliangruber — COMMENTED** [review]
Great work!

**Pull request merged**